waiter/src/waiter/scheduler.clj

```diff
@@ -190,24 +189,23 @@
 
 (defn base-url
   "Returns the url at which the service definition resides."
-  [{:keys [host port protocol]}]
+  [^String protocol ^String host port]
   (let [scheme (hu/backend-proto->scheme protocol)]
     (str scheme "://" host ":" port)))
 
 (defn end-point-url
   "Returns the endpoint url which can be queried on the url specified by the url-info map."
-  [url-info ^String end-point]
-  (str (base-url url-info)
+  [^String protocol ^String host port ^String end-point]
```

Comment:
nitpick: why does everything have a hint except `port`?

Reply:
I didn't add the type hint as I wasn't sure if it is an int or a string from the different paths where this is invoked. I can make the check and force it to `int` if you like.

Reply:
We don't have the type hint on `ServiceInstance` either, let's skip it for now.